src/app/CommandSender.h

```diff
@@ -121,25 +124,35 @@ class CommandSender final : public Messaging::ExchangeDelegate
         virtual void OnDone(CommandSender * apCommandSender) = 0;
     };
 
-    // CommandSender::ExtendedCallback::OnResponse is public SDK API, so we cannot break
+    // CommandSender::ExtendableCallback::OnResponse is public SDK API, so we cannot break
     // source compatibility for it. To allow for additional values to be added at a future
     // time without constantly changing the function's declaration parameter list, we are
-    // defining the struct AdditionalResponseData and adding that to the parameter list
-    // to allow for future extendability.
-    struct AdditionalResponseData
+    // defining the struct ResponseData and adding that to the parameter list to allow for
+    // future extendability.
+    struct ResponseData
     {
+        // The command path field in invoke command response.
+        const ConcreteCommandPath * path;
```

Comment:
Will this ever be null?  If not, can it be a reference instead of a pointer?  Ah, I see @tcarmelveilleux suggested pointers to make filling the struct easier.  But then we should clearly document which of the pointers are never null.

Reply:
Made these a reference